src/viam/sdk/rpc/server.hpp

```diff
@@ -28,13 +32,23 @@ class Server {
     /// repeated calls.
     void start();
 
-    // TODO: make `register_service` take one of our types as an arg rather than a
-    // grpc service type, and convert under the hood
```

Comment:
I tried to do this but ran into weird issues. We can't just go to registering a `ResourceServer` type unless we make it clear under the hood that it's a `grpc::Service`. But, we can't have the `ResourceServer` base type inherit from `grpc::Service` because, e.g., `MotorServer` needs to inherit from `MotorService`, and then we get an ambiguous type hierarchy inference when trying to register. We _could_ solve this by adding a `to_grpc_service` method that does appropriate typecasting or something like that, but this is pretty hacky and I don't love it. Add to this the fact that users (even those implementing new services) should never have to touch this method and it seemed not worth looking into this todo anymore.

Happy to revert if folks disagree.

Reply:
I think it's fine to just take in the `grpc::Service*`.